main.tf

```diff
@@ -128,6 +128,25 @@ resource "aws_volume_attachment" "volume-attachment" {
   volume_id    = "${element(aws_ebs_volume.volume.*.id, count.index)}"
   instance_id  = "${element(aws_instance.instance.*.id, count.index / local.num_extra_volumes)}"
   force_detach = true
+
+  # Fix for https://github.com/terraform-providers/terraform-provider-aws/issues/2084
+  provisioner "remote-exec" {
+    inline     = ["sudo poweroff"]
+    when       = "destroy"
+    on_failure = "continue"
+
+    connection {
+      host  = "${var.associate_public_ip_address ? element(aws_instance.instance.*.public_ip, count.index / local.num_extra_volumes) : element(aws_instance.instance.*.private_ip, count.index / local.num_extra_volumes)}"
+      user  = "${module.dcos-tested-oses.user}"
+      agent = true
+    }
+  }
+
+  # Make sure instance has had some time to power down before attempting volume detachment
+  provisioner "local-exec" {
```

Comment:
OS-independent delays have been proposed... https://github.com/hashicorp/terraform/issues/19243

I haven't tested this yet, but am willing to provided that I don't need to hack 10 TF repos to do it. Otherwise I'll need help testing.

Reply:
see https://jira.mesosphere.com/browse/DCOS_OSS-5311